NitroxServer/GameLogic/NitroxTimer.cs

```diff
@@ -0,0 +1,92 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using NitroxModel.Logger;
+using NitroxModel.Packets;
+using ProtoBufNet;
+
+namespace NitroxServer.GameLogic
+{
+    [Serializable]
+    [ProtoContract]
+    public class NitroxTimer : System.Timers.Timer
```

Comment:
Probably better to just wrap the `System.Timers.Timer` instead of inheriting it. You won't have to fight the existing Timer's methods/state and any possible issues with timer serialization will also be solved then.

Reply:
FYI there has been a discussion going on dropping this inheritance entirely. For posterity, let me wrap up the discussion points:

- `System.Timers.Timer` is likely not properly serializable (and/or very large), causing all kinds of trouble (applies to it being a member too);
- Timed events have a predefined (static) duration, and can (should) be solely based on the time the game has been running.
  I consider durations (like `double auroraTimer = RandomNumber(2.3d, 4d)...`) the to be static enough as well, in that it is only defined once when a new save is started, and easily persisted as a number in one of the save files.

The resulting code could do away with the `NitroxTimer` entirely, as nothing needs to be serialized. When the server loads the save it only needs to take into account the elapsed time since the save has been created, which can be stored in one of the base files (in case it isn't already). Knowing whether an event should have triggered already is as simple as checking if the trigger time is less than the elapsed game time.
Special care needs to be taken when the game is saved right as an event is supposed to fire, such that it is either processed before being saved, or processed (scheduled with a very short duration) when the save is loaded again.
Though I don't know exactly how current events hold up against this, perhaps the guarantee that it has either been triggered according to the elapsed time (thus requiring the game to be in a certain state) or scheduled to be triggered at some point in the future is enough for the game to function properly.
In much the same way this is advantageous on clients too; they can reconstruct specific parts of the game state (has the Aurora exploded, etc) by simply looking at the elapsed time received in the initial "state synchronization" packet while loading.

Keeping track of the game time on the server is incredibly easy and doesn't require a game loop either (like that game that apparently uses `update()` to `+=` the update delta to a float). The following pseudo-code would do, discussed previously as well:
```C#
public Duration ElapsedTimeThisSession => DateTime.Now() - timeServerStarted;
public Duration ElapsedTime => elapsedTimeAccordingToSave + ElapsedTimeThisSession;
```
In the event that we want the server to support pause-ing (when no clients are connected, idling while waiting for a new connection) it is only required to add `ElapsedTimeThisSession` to `elapsedTimeAccordingToSave` (and perhaps save to disk for a proper restore point), and reset `timeServerStarted` as soon as the "game world" continues simulating. The system timers need to be restarted as well.